src/client/models/generateDamage.js

```diff
@@ -0,0 +1,317 @@
+const calculateDamageArea = (array, damageX, damageY) => {
+    let x1,
+        y1,
+        x2,
+        y2,
+        theta,
+        delta = (Math.PI / 12),
+        distance,
+        pointOnCircle,
+        pointRealOnCircle = [],
+        elementToChangeFrom,
+        // setting distanceBetweenDamageSegments static as a distance between points of damaged ground
+        distanceBetweenDamageSegments = 30,
+        damageRadius = 40,
+        pointsToReplace,
+        pointsOfIntersect = [],
+        numberOfElementsToRemove;
+
+    pointsToReplace = findDamageLimits(array, damageX, damageY, damageRadius);
+    
+    pointsToReplace.map((item) => {
+        if (item[2] === 'inDamage') {
+            pointsOfIntersect.push(item);
+        }
+    });
+
+    for (let i = 1; i < pointsOfIntersect.length; i++) {
```

Comment:
not a must. in case you have time and in current situation it does not make a big difference.